src/s3_fetchers.rs

```diff
@@ -1,26 +1,31 @@
 use std::str::FromStr;
 
 use aws_sdk_s3::Client;
-use futures::stream::StreamExt;
 
 const ESTIMATED_SHARDS_COUNT: usize = 4;
 
 /// Queries the list of the objects in the bucket, grouped by "/" delimiter.
-/// Returns the list of blocks that can be fetched
-pub(crate) async fn list_blocks(
+/// Returns the list of block heights that can be fetched
+pub(crate) async fn list_block_heights(
     s3_client: &Client,
     s3_bucket_name: &str,
     start_from_block_height: crate::types::BlockHeight,
     number_of_blocks_requested: usize,
-) -> anyhow::Result<Vec<crate::types::BlockHeight>> {
+) -> Result<
+    Vec<crate::types::BlockHeight>,
+    crate::types::LakeError<aws_sdk_s3::error::ListObjectsV2Error>,
+> {
```

Comment:
Do we need this change? We never use this exact error further, we can go with anyhow::Result

Reply:
It's a preparation. I wanted to use it further, but don't want to introduce a breaking change as I have plans for releasing 0.7.0 already.